cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -1483,7 +1484,10 @@ runTransactionSubmitCmd
 
     res <- liftIO $ submitTxToNodeLocal localNodeConnInfo txInMode
     case res of
-      Net.Tx.SubmitSuccess -> liftIO $ Text.putStrLn "Transaction successfully submitted."
+      Net.Tx.SubmitSuccess -> do
+        liftIO $ Text.putStrLn "Transaction successfully submitted. Transaction hash is:"
+        let hashText = T.decodeUtf8 $ serialiseToRawBytesHex (getTxId $ getTxBody tx)
+        liftIO $ LBS.putStrLn $ Aeson.encode $ TxIdSubmission hashText
```

Comment:
The message should go to stderr, the hash should go to stdout. Otherwise it'll be a hassle to separate the two together.

Reply:
I'm not sure: do we do that already? It could surprise the user if we separate the two (because they are really both not errors!).

I wrote the JSON on a separate line so that `... | tail -n 1` works for retrieving it.

cc @CarlosLopezDeLara, in case you have insights on our conventions for printing so far.

Reply:
I think we should start doing it. `tail -n 1` is wonky, because it assumes hidden knowledge about the output format, which can change between commands.

That's pretty much way of outputting things in various unix tools. 
* `stdout` for output data
* `stderr` for diagnostic messages
If we're not doing it alredy I believe this is a good place for a precedent.
@Jimbo4350 thoughts?

Reply:
Agreed during yesterday's team meeting to do it, so it's implemented now :+1: